creator/refferalTokens/models.py

```diff
@@ -0,0 +1,55 @@
+import uuid
+from django.db import models
+from creator.studies.models import Study
+from django.contrib.auth.models import Group
+
+User = get_user_model()
+
+
+class ReferralToken(models.Model):
+    """
+    An ReferralToken will be exchanged by a new or existing user in order to
+    be added to a studies with a given role.
+    """
+
+    id = models.UUIDField(default=uuid.uuid4, unique=True, editable=False)
+
+    email = models.EmailField(
+        max_length=254,
+        unique=True,
+        help_text="The email that the token link will be sent to",
+    )
+
+    claimed = models.BooleanField(
+        default=False, help_text="If the token has been used"
+    )
+
+    is_valid = models.BooleanField(
+        default=True,
+        help_text="If the token has already been claimed or if it has expired",
+    )
+
+    study = models.ForeignKey(
```

Comment:
This is a many-many field so the token may reference multiple studies to be added to.